frontend/src/pages/projects/screens/spawner/useNotebookSizeState.ts

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import useNotebookDeploymentSize from '~/pages/projects/screens/detail/notebooks/useNotebookDeploymentSize';
+import { NotebookKind } from '~/k8sTypes';
+import { NotebookSize } from '~/types';
+import { getCustomNotebookSize } from '~/pages/projects/utils';
+import { useNotebookSize } from './useNotebookSize';
+
+export const useNotebookSizeState = (
+  existingNotebook: NotebookKind | undefined,
+): {
+  selectedSize: NotebookSize;
+  setSelectedSize: (size: string) => void;
+  sizes: NotebookSize[];
+} => {
+  const { size: notebookSize } = useNotebookDeploymentSize(existingNotebook);
+  const { sizes } = useNotebookSize();
+
+  const CUSTOM_NOTEBOOK_SIZE = getCustomNotebookSize(existingNotebook);
+
+  const allSizes = React.useMemo(() => {
+    if (existingNotebook && !sizes.some((size) => size.name === CUSTOM_NOTEBOOK_SIZE.name)) {
+      return [CUSTOM_NOTEBOOK_SIZE, ...sizes];
+    }
+    return sizes;
+  }, [sizes, existingNotebook, CUSTOM_NOTEBOOK_SIZE]);
```

Comment:
Won't this add `CUSTOM_NOTEBOOK_SIZE` into the array always even for notebooks without a custom size?

Reply:
yes, not sure why I removed the condition when I changed `useEffect` to `useMemo` 🤕